frontend/packages/ceph-storage-plugin/src/components/block-pool/block-pool-list-page.tsx

```diff
@@ -0,0 +1,292 @@
+      <TableData
+        className={blockPoolColumnInfo.usedcapacity.classes}
+        columnID={blockPoolColumnInfo.usedcapacity.id}
+      >
+        {rawCapacity}
+      </TableData>
+      <TableData
+        className={blockPoolColumnInfo.mirroringstatus.classes}
+        columnID={blockPoolColumnInfo.mirroringstatus.id}
+      >
+        {mirroringStatus ? t('ceph-storage-plugin~Enabled') : t('ceph-storage-plugin~Disabled')}
+      </TableData>
+      <TableData
+        className={blockPoolColumnInfo.overallImagehealth.classes}
+        columnID={blockPoolColumnInfo.overallImagehealth.id}
+      >
+        <Tooltip content={`${t('ceph-storage-plugin~Last synced')} ${formatedDateTime}`}>
+          <StatusIconAndText
+            title={mirroringImageHealth}
+            icon={healthStateMapping[mirroringImageHealth]?.icon}
+          />
+        </Tooltip>
+      </TableData>
+      <TableData
+        className={blockPoolColumnInfo.compressionstatus.classes}
+        columnID={blockPoolColumnInfo.compressionstatus.id}
+      >
+        {compressionStatus ? t('ceph-storage-plugin~Enabled') : t('ceph-storage-plugin~Disabled')}
+      </TableData>
+      <TableData
+        className={blockPoolColumnInfo.compressionsavings.classes}
+        columnID={blockPoolColumnInfo.compressionsavings.id}
+      >
+        {'-'}
+      </TableData>
+      <TableData className={Kebab.columnClass}>
+        <ResourceKebab
+          actions={menuActions}
+          kind={referenceFor(obj)}
+          resource={obj}
+          isDisabled={
+            obj?.metadata?.deletionTimestamp ||
+            props?.cephCluster?.metadata?.name === CEPH_EXTERNAL_CR_NAME ||
+            isDefaultPool(obj)
+          }
+          customData={{ tFunction: t }}
+        />
+      </TableData>
+    </TableRow>
+  );
+};
+
+const BlockPoolList: React.FC<BlockPoolListProps> = (props) => {
+  const { t } = useTranslation();
+
+  // Hooks
+  const [cephClusters] = useK8sWatchResource<CephClusterKind[]>(cephClusterResource);
+  const [scResources] = useK8sWatchResource<OcsStorageClassKind[]>(scResource);
+  const memoizedSC: OcsStorageClassKind[] = useDeepCompareMemoize(scResources, true);
+  const poolNames: string[] = props.data.map((pool) => pool.metadata?.name);
+  const memoizedPoolNames = useDeepCompareMemoize(poolNames, true);
+
+  // Metrics
+  const [poolRawCapacityMetrics, rawCapLoadError, rawCapLoading] = usePrometheusPoll({
+    endpoint: PrometheusEndpoint.QUERY,
+    query: getPoolQuery(memoizedPoolNames, StorageDashboardQuery.POOL_RAW_CAPACITY_USED),
+    namespace: CEPH_STORAGE_NAMESPACE,
+  });
+  const poolRawCapacity: PoolMetrics = getPerPoolMetrics(
+    poolRawCapacityMetrics,
+    rawCapLoadError,
+    rawCapLoading,
+  );
+
+  return (
+    <Table
+      {...props}
+      aria-label={t('ceph-storage-plugin~BlockPool List')}
+      Header={getHeader(t)}
+      Row={getRows}
+      customData={{
+        storageClasses: memoizedSC ?? [],
```

Comment:
Nit: Maybe we can use a type here (instead of any) for the array?

Reply:
Typecasting while receving 
https://github.com/openshift/console/pull/9252/files#diff-3706afe6bba31f923c0ee8c2adab78a372519873d3563647417ef35b93615d44R114